### PR TITLE
Add nix config

### PR DESCRIPTION
**Pull request by bwatters-r7**
- [ ] run nightly tests
- [ ] get results
- [ ] nothing explodes

### CODE REVIEW

**Comment by jmartin-r7** [comment]
Please add linking with:
* `public_configs/nightly_tests.json`
* `public_configs/supported_payloads/nix_vms.json` (or something similar)

File: public_configs/supported_payloads/Nix_vm.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  {"name":"linux/x64/meterpreter/bind_tcp", "opts": []},
+  {"name":"linux/x64/meterpreter/reverse_tcp", "opts": []},
+  {"name":"linux/x64/meterpreter_reverse_http", "opts": []},
+  {"name":"linux/x64/meterpreter_reverse_https", "opts": []},
+  {"name":"linux/x64/meterpreter_reverse_tcp", "opts": []},
+  {"name":"linux/x86/meterpreter/reverse_tcp_uuid", "opts": []},
+  {"name":"linux/x86/meterpreter_reverse_http", "opts": []},
+  {"name":"linux/x86/meterpreter_reverse_https", "opts": []},
+  {"name":"linux/x86/meterpreter_reverse_tcp", "opts": []},
+  {"name":"java/meterpreter/bind_tcp", "opts": []},
```
Comment (jmartin-r7) [inline public_configs/supported_payloads/Nix_vm.json]: Looking at this list I am inclined to think the `nightly_tests.json` segment should just include `java.json` and `python.json` reducing the list here to just `linux/*` items.

File: public_configs/nightly_tests/NixVM_meterpreter_nightly.json [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+			"HYPERVISOR_CONFIG":	"../JSON/esxi_config.json",
+			"CPE":			"cpe:/a:rapid7:metasploit:::",
+			"MSF_PATH":		"/home/msfuser/rapid7/metasploit-framework",
+			"MSF_ARTIFACT_PATH":	"/home/msfuser/rapid7/test_artifacts"
+		}
+	],
+	"TARGETS":
+	[
+		{
+		    "CPE": "cpe:/o:centos:centos:6::x64",
```
Comment (jmartin-r7) [inline public_configs/nightly_tests/NixVM_meterpreter_nightly.json]: This target format is incorrect should be 

```TARGETS":
	[
		{
		    "CPE": "cpe:/o:centos:centos:6::x64"
		},
		    "CPE": "cpe:/o:centos:centos:7::x64"
		},
		    "CPE": "cpe:/o:fedoraproject:fedora:25::x64"
		},
		    "CPE": "cpe:/o:fedoraproject:fedora:26::x64"
		},
		    "CPE": "cpe:/o:fedoraproject:fedora:27::x64"
		},
		    "CPE": "cpe:/o:fedoraproject:fedora:28::x64"
		},
		    "CPE": "cpe:/o:fedoraproject:fedora:29::x64"
		},
		    "CPE": "cpe:/o:canonical:ubuntu_linux:14.04::x64"
		}, 
		    "CPE": "cpe:/o:canonical:ubuntu_linux:16.04::x64"
		},
		    "CPE": "cpe:/o:canonical:ubuntu_linux:18.04::x64"
		}
	],
```

Comment (bwatters-r7) [reply public_configs/nightly_tests/NixVM_meterpreter_nightly.json]: I think you missed some opening braces?

Comment (bwatters-r7) [reply public_configs/nightly_tests/NixVM_meterpreter_nightly.json]: Also, that means we'll need to fix the others, but let's not do the others, here.

**Pull request merged**